Make unstruct resolvable and introduce `root.unwrap(TgpuVertexLayout)` (#781)
software-mansion/TypeGPU: apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx

| `root.unwrap(resource: TgpuBuffer<AnyData>)` | Returns a `GPUBuffer`. |
| `root.unwrap(resource: TgpuBindGroupLayout)` | Returns a `GPUBindGroupLayout`. |
| `root.unwrap(resource: TgpuBindGroup)` | Returns a `GPUBindGroup`. |
| `root.unwrap(resource: TgpuVertexLayout)` | Returns a `GPUVertexBufferLayout`. |
{/* | `root.unwrap(resource: TgpuTexture)` | Returns a `GPUTexture`. | */}
{/* | `root.unwrap(resource: TgpuReadonlyTexture \| TgpuWriteonlyTexture \| TgpuMutableTexture \| TgpuSampledTexture)` | Returns a `GPUTextureView`. | */}

:::note

> To unwrap a `TgpuVertexLayout` make sure that all its attributes are marked with the appropriate location.

Reviewer: A suggestion, this wording might suggest more that:
- It's something the dev has to do manually,
- This usually happens automatically, but it's a requirement when trying to unwrap

```suggestion
To unwrap a `TgpuVertexLayout` make sure to explicitly mark each of its attributes with the appropriate location using `d.location(...)`.
```
